tests/e2e/commontestutils/watcher.go

```diff
@@ -157,18 +157,22 @@ func GetLastModifiedTimeFromAnnotation(secret *apicorev1.Secret) (time.Time, err
 }
 
 func RotateCAManuallyWithGCM(ctx context.Context, kcpClient client.Client) error {
-	caCert := &gcertv1alpha1.Certificate{}
-	if err := kcpClient.Get(ctx, types.NamespacedName{
-		Name:      shared.CACertificateName,
-		Namespace: shared.IstioNamespace,
-	}, caCert); err != nil {
-		return fmt.Errorf("failed to get CA certificate %w", err)
-	}
-	caCert.Spec.EnsureRenewedAfter = nil
-	renew := true
-	caCert.Spec.Renew = &renew
-	if err := kcpClient.Update(ctx, caCert); err != nil {
-		return fmt.Errorf("failed to update CA certificate %w", err)
+	cert := &gcertv1alpha1.Certificate{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name:      shared.CACertificateName,
+			Namespace: shared.IstioNamespace,
+		},
+	}
+
+	patch := []byte(`{
+        "spec": {
+            "ensureRenewedAfter": null,
+            "renew": true
+        }
+    }`)
+
+	if err := kcpClient.Patch(ctx, cert, client.RawPatch(types.MergePatchType, patch)); err != nil {
+		return fmt.Errorf("failed to patch CA certificate: %w", err)
 	}
 	return nil
 }
```

Comment:
This was necessary because the update to GCM 0.18.0 has made the test flaky. We started to see the following message indicating race conditions:
```
Operation cannot be fulfilled on certificates.cert.gardener.cloud \"klm-watcher-serving\": the object has been modified; please apply your changes to the latest version and try again
```